persistent-postgresql/test/PgIntervalTest.hs

```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE EmptyDataDecls             #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE GADTs                      #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE QuasiQuotes                #-}
+{-# LANGUAGE TemplateHaskell            #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE UndecidableInstances       #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+{-# LANGUAGE DeriveAnyClass             #-}
+
+module PgIntervalTest where
+
+import PgInit
+import Control.Monad.IO.Unlift
+import Database.Persist.Postgresql (PgInterval(..))
+
+share [mkPersist sqlSettings, mkMigrate "pgIntervalMigrate"] [persistLowerCase|
+PgIntervalDb
+    interval_field PgInterval
+    deriving Eq
+    deriving Show
+|]
+
+specs :: (MonadUnliftIO m, MonadFail m) => RunDb SqlBackend m -> Spec
+specs runDb = do
+  describe "Postgres Interval tests" $ do
+    it "test for insert and read of negative intervals" $ runDb $ do
+      _ <- runMigrationSilent pgIntervalMigrate
```

Comment:
The migrations are usually stored higher up in the test tree so that individual cases don't have to deal with them.

Reply:
eg in `persistent-postgresql`, migrations are done in `test/Main.hs` at line ~100-120